### ISSUE TITLE
Remove previously build files when configuration changes

### ISSUE THREAD

**Issue by dergraaf** [modm-io/lbuild#33]
If a file has been previously been build by lbuild, it should be removed if the configuration is changed (e.g. when a module is no longer included). Currently the files stay there until they are manually deleted. Other files which have been added by a user should not be removed.

**Comment by salkinium** [comment]
YES! I've been meaning to add file hashes to the build log of both the repository files (ie. module.lb and file operation inputs) as well as the output files, so that you can have a better understanding of what exactly changed.

lbuild can then correctly warn you about files that would be overwritten.
And lbuild may also rebuild just *one* module and it's dependencies.
